src/pkgcheck/checks/reserved.py

```diff
@@ -0,0 +1,100 @@
+import re
+from .. import addons, bash, results, sources
+from . import Check
+
+
+reserved_prefixes = ('__', 'abort', 'dyn', 'prep')
+reserved_substrings = ('hook', 'paludis')  # 'ebuild' is special case
+reserved_ebuild_regex = re.compile(r'.*[^a-zA-Z]ebuild.*')
+
+
+class EclassReservedName(results.EclassResult, results.Warning):
+    """Eclass uses reserved variable or function name for package manager."""
+
+    def __init__(self, used_name: str, used_type: str, reserved_word: str, reserved_type: str, **kwargs):
+        super().__init__(**kwargs)
+        self.used_name = used_name
+        self.used_type = used_type
+        self.reserved_word = reserved_word
+        self.reserved_type = reserved_type
+
+    @property
+    def desc(self):
+        return f'{self.eclass}: {self.used_type} name "{self.used_name}" is reserved because of {self.reserved_type} "{self.reserved_word}"'
+
+
+class EclassReservedCheck(Check):
+    """Scan eclasses variables that are only allowed in certain scopes."""
```

Comment:
I think this doc comment needs updating.

Reply:
Thanks.